### PR TITLE
feat: implement struct field access for backends that support it

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds the ability to retrieve fields from struct/tuples in the DuckDB and ClickHouse backends.

### CODE REVIEW

File: ibis/backends/pandas/client.py [hunk 1]
```diff
@@ -139,9 +142,16 @@ def _infer_pandas_series_contents(s: pd.Series) -> dt.DataType:
         if inferred_dtype == 'mixed':
             # We need to inspect an element to determine the Ibis dtype
             value = s.iloc[0]
-            if isinstance(value, (np.ndarray, list, pd.Series)):
+            if isinstance(value, (np.ndarray, pd.Series, Sequence)):
                 # Defer to individual `infer` functions for these
                 return dt.infer(value)
+            elif isinstance(value, Mapping):
```
Comment (kszucs) [inline ibis/backends/pandas/client.py:148]: Shall we implement this in `expr/datatypes` instead?

File: ibis/backends/dask/tests/conftest.py [hunk 2]
```diff
@@ -19,6 +19,8 @@ def npartitions():
 
 
 class TestConf(PandasTest):
+    supports_structs = False
```
Comment (kszucs) [inline ibis/backends/dask/tests/conftest.py:22]: I see that you added the test table below, but the structs are not supported?

File: ibis/backends/pandas/execution/structs.py [hunk 3]
```diff
@@ -15,17 +15,30 @@ def execute_node_struct_field_dict(op, data, **kwargs):
...
     field = op.field
-    return data.map(operator.itemgetter(field)).rename(field)
+    return data.map(functools.partial(_safe_getter, field=field)).rename(field)
+
+
+def _safe_getter(value, field: str):
+    try:
+        return value[field]
+    except TypeError:
+        return value
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/pandas/execution/structs.py:33]: It's to handle a null value with a struct type :(

[the remark this answers is not in the document]
Comment (kszucs) [reply ibis/backends/pandas/execution/structs.py:33]: Can you handle null values explicitly instead?

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/pandas/execution/structs.py:33]: I think so.

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/pandas/execution/structs.py]: Done.

File: ibis/backends/dask/tests/conftest.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/dask/tests/conftest.py:22]: Removed the test table.

File: ibis/backends/pandas/client.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/pandas/client.py]: Done.

**Pull request merged**